data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/CountAggregateAction.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * C
+ * SPDX-Limense-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+
+import java.util.Optional;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This action will add the unique keys of each smaller Event to the overall groupState,
+ * and will create a combined Event from the groupState on concludeGroup. If smaller Events have the same keys, then these keys will be overwritten with the keys of the
+ * most recently handled Event.
+ * @since 1.3
+ */
+@DataPrepperPlugin(name = "count", pluginType = AggregateAction.class)
+public class CountAggregateAction implements AggregateAction {
+    static final String EVENT_TYPE = "event";
+    // COUNTKEY string is chosen to be unique and NOT to conflict with user provided keys
+    // This COUNTKEY is used by OTEL/Prometheus format converter processor to generate a SUMMARY metric
+    public static final String COUNTKEY = "aggr._count";
```

Comment:
I OK either way but I am not sure there is value in making it configurable. This should be internal to data prepper used to convert this to a METRIC (like SUM metric) and should never be sent out as-is.

Reply:
I am also thinking if I should directly generate metric (based on config, OTEL metric by default) instead of exposing COUNTKEY outside of this processor. We could call it `count_metric` instead of `count` and generate the metric directly. What do you think?

Reply:
Relatedly, since this is going into a map and then we generate an Event, I think this will create a single field `aggr._count` and not a nested value. Is this what we want?

Reply:
Yes, I was not intending to create nested value. It is a single field unique enough to not conflict with user's keys.

Reply:
We can have a separate OTEL or Prometheus metric converter which takes aggreated values (based on the presence of "aggr._xxxxx" field in the input events and convert them to OTEL or Prometheous metrics. I am just wondering if we should combine that separate processor into aggregate processor for efficiency.